modules/ballerina-core/src/main/java/org/ballerinalang/helpers/DeepEqualHelper.java

```diff
@@ -0,0 +1,279 @@
+                    }
+                }
+    
+                // Checking equality for string fields.
+                for (int i = 0; i < lhStructType.getFieldTypeCount()[2]; i++) {
+                    if (!lhStruct.getStringField(i).equals(rhStruct.getStringField(i))) {
+                        return false;
+                    }
+                }
+    
+                // Checking equality for boolean fields.
+                for (int i = 0; i < lhStructType.getFieldTypeCount()[3]; i++) {
+                    if (lhStruct.getBooleanField(i) != rhStruct.getBooleanField(i)) {
+                        return false;
+                    }
+                }
+    
+                // Checking equality for byte fields.
+                for (int i = 0; i < lhStructType.getFieldTypeCount()[4]; i++) {
+                    if (!Arrays.equals(lhStruct.getBlobField(i), rhStruct.getBlobField(i))) {
+                        return false;
+                    }
+                }
+    
+                // Checking equality for refs fields.
+                for (int i = 0; i < lhStructType.getFieldTypeCount()[5]; i++) {
+                    if (!isDeepEqual(lhStruct.getRefField(i), rhStruct.getRefField(i))) {
+                        return false;
+                    }
+                }
+                
+                break;
+            case TypeTags.MAP_TAG:
+                BMap lhMap = (BMap) lhVal;
+                BMap rhMap = (BMap) rhVal;
+                // Check if size is same.
+                if (lhMap.size() != rhMap.size()) {
+                    return false;
+                }
+    
+                // Check if key set is equal.
+                if (!lhMap.keySet().containsAll(rhMap.keySet())) {
+                    return false;
+                }
+    
+                List<String> keys = Arrays.stream(lhMap.keySet().toArray())
+                        .map(String.class::cast)
+                        .collect(Collectors.toList());
+                for (int i = 0; i < lhMap.size(); i++) {
+                    if (!isDeepEqual(lhMap.get(keys.get(i)), rhMap.get(keys.get(i)))) {
+                        return false;
+                    }
+                }
+                break;
+            case TypeTags.ARRAY_TAG:
+            case TypeTags.ANY_TAG:
+                // TODO: This block should ideally be in the ARRAY_TAG case only. Not ANY_TAG. #4505.
+                if (lhVal instanceof BNewArray) {
+                    BNewArray lhArray = (BNewArray) lhVal;
+                    BNewArray rhArray = (BNewArray) rhVal;
+                    if (lhArray.size() != rhArray.size()) {
+                        return false;
+                    }
+    
+                    for (int i = 0; i < lhArray.size(); i++) {
+                        if (!isDeepEqual(lhArray.getBValue(i), rhArray.getBValue(i))) {
+                            return false;
+                        }
+                    }
+                }
+                
```

Comment:
New line before the break is not consistent across all the cases. Check and remove unnecessary new lines

Reply:
Fixed with https://github.com/ballerina-lang/ballerina/pull/4532/commits/c728619b2fcdb290f1587e71e11911955ccfae3f